google/cloud/storage/internal/grpc_client.cc

```diff
@@ -457,26 +475,22 @@ StatusOr<CreateResumableSessionResponse> GrpcClient::CreateResumableSession(
   auto response = stub_->StartResumableWrite(context, *proto_request);
   if (!response.ok()) return std::move(response).status();
 
-  auto self = shared_from_this();
-  return CreateResumableSessionResponse{
-      absl::make_unique<GrpcResumableUploadSession>(
-          self, request,
-          ResumableUploadSessionGrpcParams{response->upload_id()}),
-      ResumableUploadResponse{/*.upload_session_url=*/std::string{},
-                              ResumableUploadResponse::kInProgress,
-                              /*.committed_size=*/0,
-                              /*.payload=*/absl::nullopt,
-                              /*.annotations=*/std::string{}}};
-}
-
-StatusOr<CreateResumableUploadResponse> GrpcClient::CreateResumableUpload(
-    ResumableUploadRequest const&) {
-  return Status(StatusCode::kUnimplemented, "TODO(#8621)");
+  return CreateResumableUploadResponse{response->upload_id()};
 }
 
 StatusOr<QueryResumableUploadResponse> GrpcClient::QueryResumableUpload(
-    QueryResumableUploadRequest const&) {
-  return Status(StatusCode::kUnimplemented, "TODO(#8621)");
+    QueryResumableUploadRequest const& request) {
+  OptionsSpan span(options_);
+  grpc::ClientContext context;
+  ApplyQueryParameters(context, request, "resource");
+  auto const timeout = options_.get<TransferStallTimeoutOption>();
```

Comment:
Nitty nit ... and it is only in carried-forward code ... and it doesn't look like it makes a difference in this case ... and it looks like there are similar instances, so it undoubtedly shouldn't be address here ... but, if we're going to instantiate an `OptionsSpan`, we should probably use `CurrentOptions()` for access.

Reply:
I am sure you are right, I created #8770 let's continue the conversation there.